components/editor/TimesheetEntryEditor.tsx

```diff
@@ -38,6 +38,7 @@ const DayEditor = ({
 
     const displayEntries = entries.filter((entry) => entry.date === dateStr)
     const [timesheet, setTimesheet] = useState<Timesheet>()
+    const [inEditState, setInEditState] = useState<boolean>(false)
```

Comment:
This is "half-correct". While this does indeed work well for a single entry, it behaves in a little unexpected way when there are several entries: When you click on any of the entries **all** of the entries will enter edit-mode.

What you will need instead is to have this state **per entry**. The easiest way to achieve this is to put the stuff that's approximately between lines `86` and `139` to its own component and put this kind of line into that component. Then you should render such component for each `entry` and everything will work nicely.